src/stan/mcmc/chains.hpp

```diff
@@ -254,22 +256,39 @@ namespace stan {
           var_plus += variance(chain_mean);
         Eigen::VectorXd rho_hat_t(n_samples);
         rho_hat_t.setZero();
-        double rho_hat = 0;
-        int max_t = 0;
-        for (int t = 1; (t < n_samples && rho_hat >= 0); t++) {
-          Eigen::VectorXd acov_t(chains);
-          for (int chain = 0; chain < chains; chain++) {
-            acov_t(chain) = acov(chain)(t);
+        double rho_hat_even = 1;
+        double rho_hat_odd = 0;
```

Comment:
Remove this declaration + definition. Move it both down to where it's first used, line 264.

So, it should look more like:
```
double rho_hat_even = 1;
double rho_hat_odd = 1 - (mean_var - mean(acov_t)) / var_plus;
```